api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java

```diff
@@ -293,4 +294,17 @@ public Provider getUnknownProvider() {
 		return getProviderByUuid(Context.getAdministrationService().getGlobalProperty(
 		    OpenmrsConstants.GP_UNKNOWN_PROVIDER_UUID));
 	}
+
+	@Override
+	public Provider createProviderFromUser(User user) {
+		Provider p = new Provider();
+		if (user == null || user.getPerson() == null) {
+			throw new RuntimeException("User can not be null");
+		} else if (Context.getProviderService().getProvidersByPerson(user.getPerson()).isEmpty()) {
+			p.setPerson(user.getPerson());
+		} else {
+			throw new RuntimeException("Unable to create Provider from user");
```

Comment:
I think this should just log a warning "provider already exists for user" and in this case the first provider returned by the getProvidersByPerson method should be returns.